rts/motoko-rts/src/types.rs

```diff
@@ -389,7 +399,7 @@ impl Value {
     /// In debug mode panics if the value is not a pointer or the
     /// pointed object is not a `Blob`.
     pub unsafe fn as_stream(self) -> *mut Stream {
-        debug_assert_eq!(self.tag(), TAG_BLOB);
+        debug_assert!(self.is_blob());
```

Comment:
I believe the stream would always be a `TAG_BLOB_B`? Maybe, a stricter assertion could be applied.